File: hamersapp/src/main/java/nl/ecci/hamers/meetings/SingleMeetingActivity.java
package nl.ecci.hamers.meetings;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.widget.Toolbar;
import android.widget.TextView;

import nl.ecci.hamers.MainActivity;
import nl.ecci.hamers.R;
import nl.ecci.hamers.helpers.HamersActivity;
import us.feras.mdv.MarkdownView;

import static nl.ecci.hamers.helpers.Utils.getMeeting;
import static nl.ecci.hamers.helpers.Utils.getUser;

public class SingleMeetingActivity extends HamersActivity {

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        this.setContentView(R.layout.meeting_detail);

        final Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        final ActionBar actionBar = getSupportActionBar();

        if (actionBar != null) {
            actionBar.setDisplayHomeAsUpEnabled(true);
            actionBar.setHomeButtonEnabled(true);
        }

        TextView subjectTV = (TextView) findViewById(R.id.meeting_subject);
        TextView dateTV = (TextView) findViewById(R.id.meeting_date);
        TextView userTV = (TextView) findViewById(R.id.meeting_user);
        MarkdownView agendaMV = (MarkdownView) findViewById(R.id.meeting_agenda);
        MarkdownView notesMV = (MarkdownView) findViewById(R.id.meeting_notes);

        Meeting meeting = getMeeting(MainActivity.prefs, getIntent().getIntExtra(Meeting.ID, -1));

        if (subjectTV != null && dateTV != null && userTV != null && agendaMV != null && notesMV != null) {
            subjectTV.setText(meeting.getSubject());
            dateTV.setText(MainActivity.appDF.format(meeting.getDate()));
            userTV.setText(String.format("Genotuleerd door: %s", getUser(MainActivity.prefs, meeting.getUserID()).getName()));
            agendaMV.loadMarkdown(meeting.getAgenda());
            notesMV.loadMarkdown(meeting.getNotes());
        }
    }
}

File: hamersapp/src/main/java/nl/ecci/hamers/meetings/SingleMeetingActivity.kt
package nl.ecci.hamers.meetings

import android.os.Bundle
import kotlinx.android.synthetic.main.activity_detail_item.*
import kotlinx.android.synthetic.main.stub_detail_meeting.*
import nl.ecci.hamers.BuildConfig
import nl.ecci.hamers.MainActivity
import nl.ecci.hamers.R
import nl.ecci.hamers.helpers.DataUtils
import nl.ecci.hamers.helpers.HamersActivity
import nl.ecci.hamers.helpers.Utils

class SingleMeetingActivity : HamersActivity() {

    private var meeting: Meeting? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_detail_item)

        initToolbar()

        stub_detail_item.layoutResource = R.layout.stub_detail_meeting
        stub_detail_item.inflate()

        // Intent by clicking an event in EventFragment or by clicking a link elsewhere
        val appLinkData = intent.data
        val meetingID = intent.getIntExtra(Meeting.MEETING, Utils.notFound)
        if (appLinkData != null) {
            meeting = DataUtils.getMeeting(this, Utils.getIdFromUri(appLinkData))
        } else {
            meeting = DataUtils.getMeeting(this, meetingID)
        }

        var meetingSubject = meeting?.subject
        if (BuildConfig.DEBUG) {
            meetingSubject += " (" + meeting?.id + ")"
        }
        meeting_subject.text = meetingSubject
        meeting_date.text = MainActivity.appDF.format(meeting?.date)
        meeting_user.text = String.format("Genotuleerd door: %s", DataUtils.getUser(this, meeting!!.userID).name)
        meeting_agenda.loadMarkdown(meeting?.agenda)
        meeting_notes.loadMarkdown(meeting?.notes)
    }
}
